homeassistant/components/subaru/sensor.py

```diff
@@ -272,14 +257,12 @@ def available(self) -> bool:
             return False
         return last_update_success
 
-    def get_current_value(self):
+    def get_current_value(self) -> None | int | float:
         """Get raw value from the coordinator."""
-        value = self.coordinator.data[self.vin][VEHICLE_STATUS].get(self.data_field)
-        if value in sc.BAD_SENSOR_VALUES:
-            value = None
-        if isinstance(value, str):
-            if "." in value:
-                value = float(value)
-            else:
-                value = int(value)
+        value = None
+        if isinstance(data := self.coordinator.data, dict):
```

Comment:
This will always be a dict if the coordinator has refreshed ok. We refresh before setting up the platform. The entity will be unavailable if the coordinator fails to refresh.